### ISSUE TITLE
m2t and discrete colormap

### ISSUE THREAD

**Issue by StephanMuenzberg** [matlab2tikz/matlab2tikz#573]
Hi everybody,

vor a diagram I with 20 plots I would like to use a discrete colormap instead of a normal legend. In Matlab I create this bar with "colormap and colorbar utilities" from FEX (http://www.mathworks.com/matlabcentral/fileexchange/24371-colormap-and-colorbar-utilities--jul-2014-). 

But when I export it using matlab2tikz I get only the diagram information, not the colorbar. What could I do or how do I prepare a colorbar by myself in tikz.

If needed, I could upload the Matlab code and the m2t-generated file.

![screen_shot](https://cloud.githubusercontent.com/assets/11192849/6620200/bca1cf76-c8cf-11e4-8af7-8aa3ac64f2ba.png)

**Comment by PeterPablo** [comment]
I recently had a similar issue: My goal was to plot a colorbar with three discrete gray-level values. I posted my solution over at [Tex StackExchange](http://tex.stackexchange.com/a/232710/4640).

**Comment by StephanMuenzberg** [comment]
Many thanks @PeterPablo. Do you know what's the reason why the colorbar is ignored?

**Comment by PeterPablo** [comment]
No. And to give any developer a chance to investigate, please post a minimal example of matlab code that yields that behavior. Ideally strip the code down, so that it does not use any code from FEX.

**Comment by okomarov** [comment]
> If needed, I could upload the Matlab code and the m2t-generated file.

Yes, please do (minimal matlab code should suffice).

**Comment by StephanMuenzberg** [comment]
As you wish. 

clear all
clc
close all

x = 1:3;

y = rand(3,3);

colorlist = jet(3);

figure(1)
hold on
for ii = 1:size(y,1)
    
    plot(x(:),y(ii,:),'color', colorlist(ii,:))

end

    cmfit(colorlist(:,:),[0, size(colorlist,1)],1)
    
    h = colorbar;
    
    h.Ticks = 0.5:1:3;
    keyboard
    h.TickLabels = ['1'; '2'; '3'];

**Comment by PeterPablo** [comment]
Here is a simplified deterministic example to show the issue:
```
pic=zeros(3,3);
pic(1,1) = 1;
pic(2,2) = 2;
pic(3,3) = 3;
imagesc(pic);
colorbar;
mycolormap = [1 1 1]' * (3:-1:0)/3;
colormap(mycolormap')
```
![image](https://cloud.githubusercontent.com/assets/4340267/6634808/f43e75b8-c95d-11e4-95b7-bd2714f97afd.png)
